cmd/config.go

```diff
@@ -75,6 +75,15 @@ type DBConfig struct {
 	ConnMaxIdleTime ConfigDuration
 }
 
+// DatabaseConfig is a temporary struct that acts as a receiver for
+// fields unmarshalled from the root of a component's JSON config
+// (deprecated).
+// TODO(#5275): Remove once all configs in dev, staging and prod
+// have been updated to contain the `dbconfig` field
+type DatabaseConfig struct {
+	DBConfig
```

Comment:
Instead of an embedded struct, let's try just `type DeprecatedDBConfig DBConfig`.